api/utils/aws/partition.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+// GetPartitionFromRegion get aws partition from region
+// example, region "us-east-1" corresponds to partition "aws"
+// region "cn-north-1" corresponds to partition "aws-cn"
+func GetPartitionFromRegion(region string) string {
+	var partition string
+	switch {
+	case IsCNRegion(region):
+		partition = CNPartition
+	case IsUSGovRegion(region):
+		partition = USGovPartition
+	default:
+		partition = StandardPartition
+	}
+	return partition
+}
+
+const (
```

Comment:
These are not all of the valid partitions. See https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#pkg-constants

Is there a reason why we're hardcoding all of this instead of using the `aws-sdk` that we already include?

See also https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/ and https://aws.amazon.com/blogs/developer/using-the-aws-sdk-for-gos-regions-and-endpoints-metadata/.

For example, you can get a partition from a region by using:
```
if p, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region); ok {
  partition = p.ID()
}
```

Reply:
You raise a good point, which I did previously consider.

When having a quick look at this package `api/utils/aws` I noticed there are other places where we currently only support the standard, china, and govcloud partitions, such as https://github.com/gravitational/teleport/blob/c469a34994e87eab50e1899a184a3b35a70def73/api/utils/aws/region.go#L48

I believe we also avoid importing the aws sdk into the `api` module to avoid bloating imports for api users.

I was about to start refactoring these helpers under `lib` so that I could use the SDK, and doing this properly for all partitions, and making a number of other improvements, but this all takes time and I decided to try to just accept this net improvement from the community.

Reply:
The other two partitions are "secret" partitions so unlikely we will go there. If we do, that would be awesome.

> I believe we also avoid importing the aws sdk into the api module to avoid bloating imports for api users.

👍  Yes. I'd love to include aws-sdk as we have quite some references. But aws-sdk is also a huge dependency.

Reply:
@nklaassen Makes sense! Just wanted to point it out. :-)